File: backend/internal/models/note.go
package models

import "time"

// Note représente une note opérationnelle liée à des IOCs, techniques ou CVE.
type Note struct {
	ID               uint      `json:"id"                  gorm:"primaryKey;autoIncrement"`
	CreatedAt        time.Time `json:"created_at"`
	UpdatedAt        time.Time `json:"updated_at"`
	Title            string    `json:"title"               gorm:"not null"`
	Content          string    `json:"content"             gorm:"type:text"`
	Tags             string    `json:"tags"                gorm:"type:text"` // JSON array
	LinkedIOCs       string    `json:"linked_iocs"         gorm:"type:text"` // JSON array of IDs
	LinkedTechniques string    `json:"linked_techniques"   gorm:"type:text"` // JSON array
	LinkedCVEs       string    `json:"linked_cves"         gorm:"type:text"` // JSON array
	CreatedBy        string    `json:"created_by"`
}

File: frontend/src/types/note.ts
export interface Note {
  id: number
  created_at: string
  updated_at: string
  title: string
  content: string
  tags: string       // JSON array
  linked_iocs: string        // JSON array of IDs
  linked_techniques: string  // JSON array
  linked_cves: string        // JSON array
  created_by: string
}

export interface NoteCreateRequest {
  title: string
  content?: string
  tags?: string
  linked_iocs?: string
  linked_techniques?: string
  linked_cves?: string
}

export interface NotesResponse {
  notes: Note[]
  total: number
}
